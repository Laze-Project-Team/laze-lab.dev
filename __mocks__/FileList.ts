class FileList<T> {
  _files: T[];
  length: number;
  constructor(files: T[]) {
    this._files = files;
    this.length = this._files.length;
  }
  item(index: number) {
    return this._files[index];
  }
  get [Symbol.toStringTag]() {
    return 'FileList';
  }
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.FileList = FileList;

export {};
