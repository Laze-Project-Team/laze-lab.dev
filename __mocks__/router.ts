import type { useRouter } from 'next/router';
import { beforeEach, vi } from 'vitest';

export const mockRouter = (): void => {
  vi.mock('next/router', () => ({
    useRouter: () => {
      const router: ReturnType<typeof useRouter> = {
        asPath: '/',
        back: vi.fn(),
        basePath: '/',
        beforePopState: vi.fn(),
        events: {
          on: vi.fn(),
          off: vi.fn(),
          emit: vi.fn(),
        },
        isFallback: false,
        isLocaleDomain: false,
        isPreview: false,
        isReady: true,
        locale: 'en',
        locales: ['en'],
        prefetch: vi.fn(),
        push: vi.fn(),
        query: {},
        reload: vi.fn(),
        replace: vi.fn(),
        route: '/',
        pathname: '/',
      };

      return router;
    },
  }));
};

export const mockEachRouter = (): void => {
  beforeEach(() => {
    mockRouter();
  });
};
