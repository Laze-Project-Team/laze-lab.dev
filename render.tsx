import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
