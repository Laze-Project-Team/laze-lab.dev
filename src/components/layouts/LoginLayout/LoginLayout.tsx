import { css } from '@emotion/react';
import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
import Container from '@mui/material/Container';
import { useTranslation } from 'next-i18next';
import type { FC, ReactNode } from 'react';

import { DefaultLink } from '@/components/ui/DefaultLink';
import { pagesPath } from '@/lib/$path';

export type loginLayoutProps = {
  title: ReactNode;
  children?: ReactNode;
};

export type presentialLoginLayoutProps = loginLayoutProps;

export const PresentialLoginLayout: FC<presentialLoginLayoutProps> = ({
  title,
  children,
}) => {
  const [t] = useTranslation('common');

  return (
    <Container maxWidth="sm">
      <main>
        <h1
          css={css`
            margin: 2rem 0;
            font-size: 1.5rem;
            text-align: center;
          `}
        >
          {title}
        </h1>

        <DefaultLink href={pagesPath.$url()}>
          <ArrowBackIosIcon sx={{ fontSize: '1rem' }} />
          <span>{t('backToHome')}</span>
        </DefaultLink>

        {children}
      </main>
    </Container>
  );
};

export const LoginLayout: FC<loginLayoutProps> = (props) => {
  return <PresentialLoginLayout {...props} />;
};
