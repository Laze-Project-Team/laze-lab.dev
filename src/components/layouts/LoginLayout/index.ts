export type { loginLayoutProps } from './LoginLayout';
export { LoginLayout } from './LoginLayout';
