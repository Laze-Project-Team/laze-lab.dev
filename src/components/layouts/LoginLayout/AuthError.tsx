import { Alert } from '@mantine/core';
import type { FC, ReactNode } from 'react';
import { createContext, useState } from 'react';

import { useAuthError } from '@/components/layouts/LoginLayout/useAuthError';

export type authErrorContextType = {
  error: string | null;
  setError: (error: string | null) => void;
};

export const authErrorContext = createContext<authErrorContextType | null>(
  null,
);

export type authError = {
  code: string;
  message: string;
};

export type authErrorsProviderProps = {
  children: ReactNode;
};

export const AuthErrorProvider: FC<authErrorsProviderProps> = ({
  children,
}) => {
  const [error, setError] = useState<string | null>(null);

  return (
    <authErrorContext.Provider value={{ error, setError }}>
      {children}
    </authErrorContext.Provider>
  );
};

export const AuthError: FC = () => {
  const { error, setError } = useAuthError();

  return <>{error && <Alert onClose={() => setError(null)}>{error}</Alert>}</>;
};
