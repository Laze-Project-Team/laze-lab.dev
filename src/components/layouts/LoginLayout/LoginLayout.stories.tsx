import type { ComponentMeta, ComponentStory } from '@storybook/react';

import { PresentialLoginLayout } from './LoginLayout';

export default {
  title: 'layouts/LoginLayout',
  component: PresentialLoginLayout,
} as ComponentMeta<typeof PresentialLoginLayout>;

const Template: ComponentStory<typeof PresentialLoginLayout> = (props) => (
  <PresentialLoginLayout {...props} />
);

export const Primary = Template.bind({});
Primary.args = {
  title: 'Here is the title',
  children: 'Here is the children',
};
