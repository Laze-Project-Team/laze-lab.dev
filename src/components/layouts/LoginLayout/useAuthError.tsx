import { useContext } from 'react';

import type { authError } from './AuthError';
import { authErrorContext } from './AuthError';

export type authErrorHandlers = {
  error: string | null;
  setError: (errorMessage: string | null) => void;
  handleError: (error: authError) => void;
};

export const useAuthError = (): authErrorHandlers => {
  const authCtx = useContext(authErrorContext);

  if (authCtx === null) {
    throw new Error('Please wrap the component with AuthErrorProvider');
  }

  const { error, setError } = authCtx;

  return {
    error,
    setError: (errorMessage) => {
      setError(errorMessage);
    },
    handleError: (err) => {
      setError(err.message);
    },
  };
};
