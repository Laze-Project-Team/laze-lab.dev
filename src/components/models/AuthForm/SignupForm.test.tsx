import '@testing-library/jest-dom';

import { fireEvent, screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { mockEachRouter } from '/__mocks__/router';
import { AuthErrorProvider } from '@/components/layouts/LoginLayout/AuthError';
import { render } from '@/lib/test/render';

import { SignupForm } from '.';

mockEachRouter();

it('should be rendered correctly', () => {
  const { container } = render(<SignupForm />, { wrapper: AuthErrorProvider });
  expect(container.firstChild).toMatchSnapshot();
});

it('should have "form" role', () => {
  render(<SignupForm />, { wrapper: AuthErrorProvider });
  expect(screen.getByRole('form')).toBeInTheDocument();
});

it('should work as login form', () => {
  const form = render(<SignupForm />, { wrapper: AuthErrorProvider });

  const emailInput = form.getByRole('textbox', {
    name: 'form.email.label',
  }) as HTMLInputElement;
  const emailValue = 'aaa-bbb-ccc@example.com';
  fireEvent.change(emailInput, {
    target: { value: emailValue },
  });
  expect(emailInput.value).toBe(emailValue);

  const passwordInput = form.getByRole('textbox', {
    name: 'form.password.label',
  }) as HTMLInputElement;
  const passwordValue = 'this is password';
  fireEvent.change(passwordInput, {
    target: { value: passwordValue },
  });
  expect(passwordInput.value).toBe(passwordValue);
});
