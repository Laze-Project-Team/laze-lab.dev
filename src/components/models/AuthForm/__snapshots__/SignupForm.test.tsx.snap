// Vitest Snapshot v1

exports[`should be rendered correctly 1`] = `
<form
  aria-label="form.label"
  role="form"
>
  <div
    class="css-1p5q5e5-MuiStack-root"
  >
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r0:"
        id=":r0:-label"
      >
        form.label.email
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-describedby=":r0:-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          defaultvalue=""
          id=":r0:"
          name="email"
          type="text"
        />
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
        id=":r0:-helper-text"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </p>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for=":r1:"
        id=":r1:-label"
      >
        form.label.password
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
      >
        <input
          aria-describedby=":r1:-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          defaultvalue=""
          id=":r1:"
          name="password"
          type="text"
        />
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
        id=":r1:-helper-text"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </p>
    </div>
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          defaultvalue="false"
          name="agreement"
          type="checkbox"
          value="false"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        <a
          class="css-hn4o6p-PresentialStyledLink-PresentialStyledLink-PresentialDefaultLink-PresentialDefaultLink"
          href="/terms"
        />
        <a
          class="css-hn4o6p-PresentialStyledLink-PresentialStyledLink-PresentialDefaultLink-PresentialDefaultLink"
          href="/privacy"
        />
      </span>
    </label>
    <p
      class="MuiFormHelperText-root css-1d1r5q-MuiFormHelperText-root"
    >
      <a
        class="css-hn4o6p-PresentialStyledLink-PresentialStyledLink-PresentialDefaultLink-PresentialDefaultLink"
        href="/terms"
      />
      <a
        class="css-hn4o6p-PresentialStyledLink-PresentialStyledLink-PresentialDefaultLink-PresentialDefaultLink"
        href="/privacy"
      />
    </p>
    <p
      class="MuiFormHelperText-root Mui-error css-1d1r5q-MuiFormHelperText-root"
    >
      <span
        class="notranslate"
      >
        ​
      </span>
    </p>
    <button
      aria-disabled="true"
      class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      form.label.submit
    </button>
  </div>
</form>
`;
